src/lib/Cashlink.ts

```diff
@@ -118,28 +121,33 @@ class Cashlink {
         public timestamp: number = Math.floor(Date.now() / 1000),
         public contactName?: string, /** unused for now */
     ) {
-        this._network = new Promise((resolve) => {
+        const networkPromise = new Promise<NetworkClient>((resolve) => {
+            // Safe resolver function for when the network client gets assigned
             this._networkClientResolver = resolve;
         });
+        this._getNetwork = () => networkPromise;
 
         if (value) this.value = value;
         if (message) this.message = message;
 
         this._immutable = !!(value || message);
 
-        this._network.then((network: NetworkClient) => {
-            // value will be updated as soon as we have consensus (in _onPotentialBalanceChange)
-            // and a Cashlink.Event.CONFIRMED_BALANCE_CHANGE event gets fired
+        this._getNetwork().then((network: NetworkClient) => {
+            const userFriendlyAddress = this.address.toUserFriendlyAddress();
+
+            // When not yet established, the balance will be updated by the nano-api as soon
+            // as we have consensus, because subscribing (below) triggers a balance check.
             if (network.consensusState === 'established') {
-                this.getAmount(); // Updates _currentBalance internally
+                network.getBalance(userFriendlyAddress).then(this._onBalancesChanged.bind(this));
             }
```

Comment:
[Optional]

Would this work correctly if the address has been subscribed before (e.g. another Cashlink instance has been generated before for this Cashlink) and consensus had already been reached but then been lost and we create this second instance while the consensus is still lost?
Instead of checking for the consensus state here, would be maybe need to check whether the address is already subscribed?

But as this is quite a corner case it also doesn't necessarily have to be handled.

Reply:
The consensusPromise gets recreated in the nano-api when consensus is lost, so new subscriptions while the consensus is lost behave just as they would when subscribing at page load.

Reply:
Ohhh... you mean because the balance change doesn't trigger because the balance stays the same? Hmm... yes, I believe this fails.

Reply:
I think the only way to fix this is to make a call to `subscribe()` always trigger a balance event for those addresses, no matter if balance has changed or not.

Reply:
If https://github.com/nimiq/nano-api/pull/9 gets merged, this manual balance check can be removed.